lib/defines.h

```diff
@@ -279,21 +279,8 @@ extern char *strdup ();
 #endif
 
 /* Maximum length of usernames */
-#ifdef HAVE_UTMPX_H
-# include <utmpx.h>
-# define USER_NAME_MAX_LENGTH (sizeof (((struct utmpx *)NULL)->ut_user))
-#else
-# include <utmp.h>
-# ifdef HAVE_STRUCT_UTMP_UT_USER
-#  define USER_NAME_MAX_LENGTH (sizeof (((struct utmp *)NULL)->ut_user))
-# else
-#  ifdef HAVE_STRUCT_UTMP_UT_NAME
-#   define USER_NAME_MAX_LENGTH (sizeof (((struct utmp *)NULL)->ut_name))
-#  else
-#   define USER_NAME_MAX_LENGTH 32
-#  endif
-# endif
-#endif
+#include <utmpx.h>
+#define USER_NAME_MAX_LENGTH (sizeof (((struct utmpx *)NULL)->ut_user))
```

Comment:
The assumption of the existence of `utmpx.h` is more complicated than your current patch. I would recommend you to look for the `USE_UTMPX` macro, as I think you can remove much more code.

Reply:
There's the `enable_utmpx` configuration option.  I don't know what it was for.  But yes, we can probably wipe it.

I'll do it, and if anyone complains, it better voices soon :)

```
$ grep -rn enable_utmpx
configure.ac:235:	 yes) enable_utmpx="yes" ;;
configure.ac:236:	  no) enable_utmpx="no" ;;
configure.ac:239:	[enable_utmpx="no"]
configure.ac:700:if test "$enable_utmpx" = "yes"; then
```

Reply:
Oh, I see. utmpx is not part of the POSIX standard. You can disregard my comment.

Reply:
<https://pubs.opengroup.org/onlinepubs/009695399/>